lib/ansible/modules/network/cloudengine/ce_ospf.py

```diff
@@ -0,0 +1,996 @@
+                    return False
+                if int(each_num) > 255:
+                    return False
+            return True
+
+        return False
+
+    def convert_ip_to_network(self):
+        """convert ip to subnet address"""
+
+        ip_list = self.addr.split('.')
+        mask_list = self.get_wildcard_mask().split('.')
+
+        for i in range(len(ip_list)):
+            ip_list[i] = str((int(ip_list[i]) & (~int(mask_list[i]))) & 0xff)
+
+        self.addr = '.'.join(ip_list)
+
+    def check_params(self):
+        """Check all input params"""
+
+        # process_id check
+        if not self.process_id.isdigit():
+            self.module.fail_json(msg="Error: process_id is not digit.")
+        if int(self.process_id) < 1 or int(self.process_id) > 4294967295:
+            self.module.fail_json(
+                msg="Error: process_id must be an integer between 1 and 4294967295.")
+
+        if self.area:
+            # area check
+            if self.area.isdigit():
+                if int(self.area) < 0 or int(self.area) > 4294967295:
+                    self.module.fail_json(
+                        msg="Error: area id (Integer) must be between 0 and 4294967295.")
+
+            else:
+                if not self.is_valid_v4addr(self.area):
+                    self.module.fail_json(msg="Error: area id is invalid.")
+
+            # area network check
+            if self.addr:
+                if not self.is_valid_v4addr(self.addr):
+                    self.module.fail_json(
+                        msg="Error: network addr is invalid.")
+                if not self.mask:
+                    self.module.fail_json(
+                        msg="Error: network mask must be set.")
+                if not self.mask.isdigit():
+                    self.module.fail_json(
+                        msg="Error: network mask is not digit.")
+                if int(self.mask) < 0 or int(self.mask) > 32:
+                    self.module.fail_json(
+                        msg="Error: network mask is invalid.")
+
+            # area authentication check
+            if self.state == "present" and self.auth_mode:
+                if self.auth_mode == "simple":
+                    if self.auth_text_simple and len(self.auth_text_simple) > 8:
+                        self.module.fail_json(
+                            msg="Error: auth_text_simple is not in the range from 1 to 8.")
+                if self.auth_mode in ["hmac-sha256", "hmac-sha256", "md5"]:
+                    if self.auth_key_id and not self.auth_text_md5:
+                        self.module.fail_json(
+                            msg='Error: auth_key_id and auth_text_md5 should be set at the same time.')
+                    if not self.auth_key_id and self.auth_text_md5:
+                        self.module.fail_json(
+                            msg='Error: auth_key_id and auth_text_md5 should be set at the same time.')
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```